### PR TITLE
DDF-3676 Refactor the registry IdentificationPlugin to query the catalog every time instead of caching registry ids

### PR DESCRIPTION
**Pull request by clockard**
#### What does this PR do?
Refactor the registry IdentificationPlugin to query the catalog every time instead of caching registry ids
#### Who is reviewing it? 
@mcalcote @vinamartin @emmberk
#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@coyotesqrl
@figliold
#### How should this be tested? (List steps with links to updated documentation)
Install 2 DDFs with registries
Publish/subscribe DDF2 to DDF1
Add a secondary local node to DDF1 and verify DDF2 gets the update
Use solr admin ui on DDF2 to lookup the metacard id for the secondary local node (need the id for the registry and remote.registry entries)
On DDF2 command line run `catalog:remove -p <id1> <id2>`
Watch the logs on DDF2 and make sure there are no `Can't create duplicate registry entries` errors in the log.
Verify that the secondary node gets created again on DDF2

#### What are the relevant tickets?
[DDF-3676](https://codice.atlassian.net/browse/DDF-3676)

### CODE REVIEW

File: catalog/spatial/registry/registry-identification-plugin/src/main/java/org/codice/ddf/registry/identification/RegistryIdRetriever.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
+/**
```
Comment (clockard) [inline catalog/spatial/registry/registry-identification-plugin/src/main/java/org/codice/ddf/registry/identification/RegistryIdRetriever.java]: This isn't a new class, just renamed RegistryIdPostIngestPlugin

File: catalog/spatial/registry/registry-identification-plugin/src/test/java/org/codice/ddf/registry/identification/RegistryIdRetrieverTest.java [hunk 2]
```diff
@@ -0,0 +1,133 @@
+/**
```
Comment (clockard) [inline catalog/spatial/registry/registry-identification-plugin/src/test/java/org/codice/ddf/registry/identification/RegistryIdRetrieverTest.java]: Not new just a rename

File: catalog/spatial/registry/registry-identification-plugin/src/main/java/org/codice/ddf/registry/identification/RegistryIdRetriever.java [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+
+  private FilterBuilder filterBuilder;
+
+  private Security security;
+
+  public RegistryIdRetriever() {
+    security = Security.getInstance();
+  }
+
+  public RegistryIdRetriever(Security security) {
```
Comment (emmberk) [inline catalog/spatial/registry/registry-identification-plugin/src/main/java/org/codice/ddf/registry/identification/RegistryIdRetriever.java]: Is this method `@VisibleForTesting`?

File: catalog/spatial/registry/registry-identification-plugin/src/test/java/org/codice/ddf/registry/identification/IdentificationPluginTest.java [hunk 4]
```diff
@@ -72,6 +76,8 @@
 
   private MetacardImpl sampleData;
 
+  @Mock private RegistryIdRetriever registryIdRetriever;
```
Comment (emmberk) [inline catalog/spatial/registry/registry-identification-plugin/src/test/java/org/codice/ddf/registry/identification/IdentificationPluginTest.java]: Is this used anywhere outside of the `setUp()` method? Can it be just defined in that method instead?
```java
    final RegistryIdRetriever registryIdRetriever = mock(RegistryIdRetriever.class);
    when(registryIdRetriever.getRegistryIdInfo())
        .thenReturn(new RegistryIdRetriever.RegistryIdInfo(Collections.emptyList()));
    identificationPlugin.setRegistryIdRetriever(registryIdRetriever);
```

File: catalog/spatial/registry/registry-identification-plugin/src/test/java/org/codice/ddf/registry/identification/RegistryIdRetrieverTest.java [hunk 5]
```diff
@@ -0,0 +1,133 @@
...
+    when(security.runAsAdminWithException(any(PrivilegedExceptionAction.class)))
+        .thenAnswer(invocation -> ((PrivilegedExceptionAction) invocation.getArguments()[0]).run());
+    when(security.runWithSubjectOrElevate(any(Callable.class)))
+        .thenAnswer(invocation -> ((Callable) invocation.getArguments()[0]).call());
+    when(framework.query(any(QueryRequest.class))).thenReturn(response);
+    RegistryIdRetriever.RegistryIdInfo idInfo = registryIdRetriever.getRegistryIdInfo();
+    assertThat(idInfo.getRegistryIds().size(), equalTo(1));
+    assertThat(idInfo.getRegistryIds().iterator().next(), equalTo("regId"));
+    assertThat(idInfo.getLocalRegistryIds().size(), equalTo(0));
+    assertThat(idInfo.getRemoteMetacardIds().size(), equalTo(0));
```
Comment (emmberk) [inline catalog/spatial/registry/registry-identification-plugin/src/test/java/org/codice/ddf/registry/identification/RegistryIdRetrieverTest.java]: I know this wasn't you changes, but some hamcrest methods could be used for these unit tests:

```java
...
import static org.hamcrest.Matchers.contains;
import static org.hamcrest.Matchers.is;
import static org.hamcrest.collection.IsEmptyCollection.empty;
...
  @Test
  public void testGetRegistryIdInfoNormalRegistryMetacard() throws Exception {
  ...
    RegistryIdRetriever.RegistryIdInfo idInfo = registryIdRetriever.getRegistryIdInfo();
    assertThat(idInfo.getRegistryIds(), contains("regId"));
    assertThat(idInfo.getLocalRegistryIds(), is(empty()));
    assertThat(idInfo.getRemoteMetacardIds(), is(empty()));
  }

  @Test
  public void testGetRegistryIdInfoInternalRegistryMetacard() throws Exception {
    ...
    RegistryIdRetriever.RegistryIdInfo idInfo = registryIdRetriever.getRegistryIdInfo();
    assertThat(idInfo.getRegistryIds(), contains("remoteMcardId"));
    assertThat(idInfo.getLocalRegistryIds(), is(empty()));
    assertThat(idInfo.getRemoteMetacardIds(), is(empty()));
  }

  @Test
  public void testGetRegistryIdInfoLocalRegistryMetacard() throws Exception {
  ...
    RegistryIdRetriever.RegistryIdInfo idInfo = registryIdRetriever.getRegistryIdInfo();
    assertThat(idInfo.getRegistryIds(), contains("regId"));
    assertThat(idInfo.getLocalRegistryIds(), contains("regId"));
    assertThat(idInfo.getRemoteMetacardIds(), is(empty()));
  }
...
```

**Comment by clockard** [comment]
build now

**Comment by clockard** [comment]
build now

**Comment by peterhuffer** [comment]
Successful hero. Verified steps and that no logs were generated when deleting registry metacards, and that the registry metacards were recreated.

**Comment by clockard** [comment]
build now

**Comment by clockard** [comment]
build now

**Comment by clockard** [comment]
Sonarqube issue is outside the scope of this change.

**Pull request merged**